responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -510,6 +566,33 @@ def _validate_rai_insights_input_parameters(
                 "Expecting pandas DataFrame for train and test."
             )
 
+    def _validate_feature_metadata(self, feature_metadata, train, task_type):
+        if feature_metadata is not None:
+            if not isinstance(feature_metadata, FeatureMetadata):
+                raise UserConfigValidationException(
+                    "Expecting type FeatureMetadata but got "
+                    f"{type(feature_metadata)}")
+
+            feature_names = list(train.columns)
+            feature_metadata.validate_feature_metadata_with_user_features(
+                feature_names)
+
+            if task_type != ModelTask.FORECASTING:
+                if feature_metadata.time_series_id_features:
+                    raise UserConfigValidationException(
+                        "The specified metadata time_series_id_features "
+                        "is only supported for the forecasting task type.")
+
+                if feature_metadata.datetime_features:
+                    raise UserConfigValidationException(
+                        "The specified metadata datetime_features "
+                        "is only supported for the forecasting task type.")
```

Comment:
Does this mean we don't support support datetime features in classification and regression scenario?

Reply:
Correct. That is exactly as before this PR, we're just explicit about it.

Reply:
Why can't we use datetime_features in classification and regression? It is not the question of supporting now but why do we need this explicit exception?

Reply:
I don't quite follow. Either you support it, or you don't. If I were to specify a datetime column right now, we will error out. We're simply preserving that behavior for all cases except `task_type="forecasting"` with exactly 1 datetime feature. In all other cases we make users aware that they are configuring a scenario that isn't yet supported.